src/formatters/utils/sortResults.ts

```diff
@@ -1,13 +1,54 @@
 import { IRuleResult } from '../../types';
 
-export const sortResults = (results: IRuleResult[]) => {
-  return [...results].sort((resultA, resultB) => {
-    const diff = resultA.range.start.line - resultB.range.start.line;
+const safeCastToString = (code: string | number | undefined): string => {
+  if (code === undefined) {
+    throw new Error('Unexpected undefined code');
+  }
+
+  if (typeof code === 'number') {
+    throw new Error('Unexpected numbered type code');
+  }
+
+  return code;
+};
+
+const ARTIFICIAL_ROOT = 'root';
```

Comment:
Let's have a symbol here or something really unique. Root can be a potential source

Reply:
@P0lip Fixed. However, I can't help but feeling slightly displeased with result. The `toString()` call isn't really self explicit with regards to the uniqueness intent.

Moreover, although quite unlikely, nothing prevents a caller from feeding `source` with a string containing `Symbol(root)`.

Thoughts?

Reply:
Let `safeSource` return a symbol or string. If there is a symbol returned, shift the item to the left (return -1).

Reply:
@P0lip I went a different way. Do you like it better?